### ISSUE TITLE
Using httpx as go code

### ISSUE THREAD

**Issue by ehsandeep** [projectdiscovery/httpx#592]
### Please describe your feature request:
A document including examples to use httpx into go projects + integration test

### Describe the use case of this feature:
As a developer, I wanted to integrate httpx into my go project to utilize all the features of httpx.

Requested by Discord user.

**Comment by jimen0** [comment]
Gave this a try, @ehsandeep. Here's the minimal example I created.

```go
package main

import (
	"github.com/projectdiscovery/httpx/runner"
)

func main() {
	opts := &runner.Options{
		Methods:   "GET",
		InputFile: "/dev/stdin",
	}

	r, err := runner.New(opts)
	if err != nil {
		panic(err)
	}
	defer r.Close()

	r.RunEnumeration()
}
```

```go
$ echo "example.com" | go run main.go
https://example.com
```

Something I noticed is the `httpx/runner.Options` public API is quite hard to understand and use based on the docs at https://pkg.go.dev/github.com/projectdiscovery/httpx@v1.2.1/runner#Options.

`runner.New` requires an Options struct which has **tens** of attributes but doesn't specify which ones are required and which ones can be skipped. https://pkg.go.dev/github.com/projectdiscovery/httpx@v1.2.1/runner#New

It might be a great idea to rework the Options and Runner APIs first. For example, a question that came to my mind is: Is the `r.Close` call actually needed? Should it be called even if `RunEnumeration` is never called?
